core/locks_hasher.go

```diff
@@ -0,0 +1,61 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"hash"
+	"hash/fnv"
+)
+
+const (
+	DefaultLockConcurrency = 1024
+	DefaultLockIdentifier  = "default-lock"
```

Comment:
With DefaultLockIdentifier we may be getting a common lock, but it may still allow concurrent requests to access the same key with different lock. wdyt?